compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -210,14 +215,26 @@ private BindingGraph create(
         explicitBindingsBuilder.add(provisionBindingFactory.forComponent(componentDependency));
         List<ExecutableElement> dependencyMethods =
             ElementFilter.methodsIn(elements.getAllMembers(componentDependency));
```

Comment:
Scratch that, looks like I mistakenly was picking up `SimpleType.A`'s `@Inject` constructor`